src/lib/route.js

```diff
@@ -187,20 +242,67 @@ function dataToRoute (data, currentEpoch) {
     additionalInfo: data.additional_info,
     targetPrefix: data.target_prefix,
     addedDuringEpoch: currentEpoch
-  })
+  }, data.paths)
 }
 
 /**
- * @param {Array} listA
- * @param {Array} listB
- * @returns {Integer} the number of items that listA and listB share
+ * @param {Route} routeA
+ * @param {Route} routeB
+ * @returns {Boolean} whether routeA and routeB can join without forming a loop
  */
-function intersect (listA, listB) {
-  let common = 0
-  for (const itemA of listA) {
-    if (listB.indexOf(itemA) !== -1) common++
+function canJoin (routeA, routeB) {
+  const visited = {}
```

Comment:
I think this variable should be moved closer to where it's being used, for readability